haystack/document_stores/weaviate.py

```diff
@@ -1382,31 +1382,47 @@ def _delete_index(self, index: str):
             self.weaviate_client.schema.delete_class(index)
             logger.info("Index '%s' deleted.", index)
 
-    def delete_labels(self):
+    def delete_labels(
+        self,
+        index: Optional[str] = None,
+        ids: Optional[List[str]] = None,
+        filters: Optional[Dict[str, Union[Dict, List, str, int, float, bool]]] = None,
+        headers: Optional[Dict[str, str]] = None,
+    ):
         """
         Implemented to respect BaseDocumentStore's contract.
 
         Weaviate does not support labels (yet).
         """
         raise NotImplementedError("Weaviate does not support labels (yet).")
 
-    def get_all_labels(self):
+    def get_all_labels(
+        self,
+        index: Optional[str] = None,
+        filters: Optional[Dict[str, Union[Dict, List, str, int, float, bool]]] = None,
+        headers: Optional[Dict[str, str]] = None,
+    ) -> List[Label]:
         """
         Implemented to respect BaseDocumentStore's contract.
 
         Weaviate does not support labels (yet).
         """
         raise NotImplementedError("Weaviate does not support labels (yet).")
 
-    def get_label_count(self):
+    def get_label_count(self, index: Optional[str] = None, headers: Optional[Dict[str, str]] = None) -> int:
         """
         Implemented to respect BaseDocumentStore's contract.
 
         Weaviate does not support labels (yet).
         """
         raise NotImplementedError("Weaviate does not support labels (yet).")
 
-    def write_labels(self):
+    def write_labels(
+        self,
+        labels: Union[List[Label], List[dict]],
+        index: Optional[str] = None,
+        headers: Optional[Dict[str, str]] = None,
+    ):
```

Comment:
This method should `raise NotImplementedError("Weaviate does not support labels (yet).")` instead of just `pass` for consistency with the other methods. Only then can users easily notice that something is wrong when they call `write_labels()` on a `WeaviateDocumentStore`. Or is there a good reason to have `pass` here?